### PR TITLE
Expanded Costing Tests

### PR DESCRIPTION
**Pull request by bpaul4**
## Summary/Motivation:
- A large number of tests for the UKy REE Costing Framework
- Associated bug fixes

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
This PR will need some conflicts resolved once #22 is merged, @bpaul4 will handle this once #22 is in.

**Comment by bpaul4** [comment]
I'm pleasantly surprised the main coverage is so high, and happy that these tests increase it 🙌 The one line not covered is a failsafe for if the recovery rate argument has no units, which shouldn't happen if users specify their variables properly. I can add a small test just to cover it if I'm addressing other comments, but in my opinion it's not critical if that's the only thing holding up this PR.

**Review by MarcusHolly — APPROVED** [review]
Thanks Brandon - LGTM

**Comment by MarcusHolly** [comment]
> I'm pleasantly surprised the main coverage is so high, and happy that these tests increase it 🙌 The one line not covered is a failsafe for if the recovery rate argument has no units, which shouldn't happen if users specify their variables properly. I can add a small test just to cover it if I'm addressing other comments, but in my opinion it's not critical if that's the only thing holding up this PR.

I don't think it's critical, so I'll let you merge the PR whenever you're ready @bpaul4

File: src/prommis/uky/costing/ree_plant_capcost.py
Comment (agarciadiego) [inline src/prommis/uky/costing/ree_plant_capcost.py:846]: will this stay commented out?

**Review by agarciadiego — APPROVED** [review]
LGTM just a question about commented-out code.

File: src/prommis/uky/costing/ree_plant_capcost.py
Comment (bpaul4) [reply src/prommis/uky/costing/ree_plant_capcost.py:846]: Yes, I commented this out when I changed the recovery rate calculations from per-hour to per-year. I will remove these lines.

**Pull request merged**
